staging/src/k8s.io/apiserver/pkg/util/flowcontrol/apf_controller.go

```diff
@@ -831,6 +831,11 @@ func (cfgCtlr *configController) startRequest(ctx context.Context, rd RequestDig
 	plState := cfgCtlr.priorityLevelStates[plName]
 	if plState.pl.Spec.Type == flowcontrol.PriorityLevelEnablementExempt {
 		klog.V(7).Infof("startRequest(%#+v) => fsName=%q, distMethod=%#+v, plName=%q, immediate", rd, selectedFlowSchema.Name, selectedFlowSchema.Spec.DistinguisherMethod, plName)
+
+		// TODO: this is a quick and dirty fix. workEstimator does two things, it estimates work
+		//  and also notes the classification of the request. work estimation is not applicable to
+		//  an exempt request, only noting is.
+		workEstimator(selectedFlowSchema, plState.pl, "")
```

Comment:
@MikeSpreitzer I can change `Handle` to take these separately:

```
Handle(ctx context.Context,
		requestDigest RequestDigest,
		noteFn func(fs *flowcontrol.FlowSchema, pl *flowcontrol.PriorityLevelConfiguration, flowDistinguisher string),
                workEstimator func() fcrequest.WorkEstimate
               ...
```

Let me know if this okay with you.

Reply:
@tkashem : If I understand your suggestion correctly, you are agreeing with https://github.com/kubernetes/kubernetes/issues/106826#issuecomment-987097507 and proposing to replace the one function parameter of Handle that both consumes the classification and does the work estimation with two function parameters that separate those two jobs.  If so then I agree.